sunkit_image/utils/utils.py

```diff
@@ -173,3 +182,439 @@ def get_radial_intensity_summary(
+        flux = np.sum(np.maximum(flux_, 0.)) / np.float32(nlen)
+
+        if flux > flux_max:
+            flux_max = flux
+            al[ip + 1] = al[ip] + sign_dir * (step / rad_i)
+            ir[ip+1] = ib
+            al_mid = (al[ip]+al[ip+1]) / 2.
+            xl[ip+1] = xl[ip] + step * np.float32(np.cos(al_mid + np.pi * idir))
+            yl[ip+1] = yl[ip] + step * np.float32(np.sin(al_mid + np.pi * idir))
+            ix_ip = min(max(int(xl[ip + 1] + 0.5), 0), nx - 1)
+            iy_ip = min(max(int(yl[ip + 1] + 0.5), 0), ny - 1)
+            zl[ip + 1] = image[ix_ip, iy_ip]
+
+    # Attempt to optimize the code by vectorizing. Not succeessful as of yet due to
+    # three dimensional search space and also the readability of the code is affected
+    
+    # nb = 30
+    # step = 1
+    # nx, ny = image.shape
+
+    # s_loop = step * np.arange(nlen, dtype=np.float32).reshape((-1, 1))
+
+    # # This denotes loop tracing in forward direction
+    # if idir == 0:
+    #     sign_dir = +1
+
+    # # This denotes loop tracing in backward direction
+    # if idir == 1:
+    #     sign_dir = -1
+    
+    # # `ib1` and `ib2` decide the range of radius in which the next point is to be searched
+    # if ip == 0:
+    #     ib1 = 0
+    #     ib2 = nb-1
+    # if ip >= 1:
+    #     ib1 = int(max(ir[ip] - 1, 0))
+    #     ib2 = int(min(ir[ip] + 1, nb-1))
+
+    # rad_i = rmin / (-1. + 2. * np.arange(ib1, ib2 + 1, dtype=np.float32) / np.float32(nb - 1)).reshape((1, -1))
+    
+    # beta0 = al[ip] + np.float32(np.pi / 2)
+
+    # # Finding the assumed centre of the loop
+    # xcen = xl[ip] + rmin * np.float32(np.cos(beta0))
+    # ycen = yl[ip] + rmin * np.float32(np.sin(beta0))
+
+    # xcen_i = xl[ip] + (xcen - xl[ip]) * (rad_i / rmin)
+    # ycen_i = yl[ip] + (ycen - yl[ip]) * (rad_i / rmin)
+
+    # beta_i = beta0 + sign_dir * np.float32(np.matmul(s_loop, 1 / rad_i))
+
+    # rad_i = rad_i.reshape(-1)
+
+    # x_pos = xcen_i - rad_i[:, None, None] * np.float32(np.cos(beta_i))
+    # y_pos = ycen_i - rad_i[:, None, None] * np.float32(np.sin(beta_i))
+
+    # # Taking the ceil as images can be indexed by pixels
+    # ix = np.int_(x_pos + 0.5)
+    # iy = np.int_(y_pos + 0.5)
+
+    # # All the coordinate values should be within the input range
+    # ix = np.clip(ix, 0, nx - 1)
+    # iy = np.clip(iy, 0, ny - 1)
+
+    # # Calculating the mean flux at possible x and y locations
+    # flux_ = image[ix, iy]
+
+    # flux = flux_.sum(-1).sum(-1) / np.float32(nlen)
+
+    # v = np.argmax(flux)
+
+    # al[ip + 1] = al[ip] + sign_dir * (step / rad_i[v])
+    # ir[ip+1] = ib1 + v
+
+    # al_mid = (al[ip]+al[ip+1]) / 2.
+
+    # xl[ip+1] = xl[ip] + step * np.float32(np.cos(al_mid + np.pi * idir))
+    # yl[ip+1] = yl[ip] + step * np.float32(np.sin(al_mid + np.pi * idir))
+    # ix_ip = min(max(int(xl[ip + 1] + 0.5), 0), nx - 1)
+    # iy_ip = min(max(int(yl[ip + 1] + 0.5), 0), ny - 1)
+    # zl[ip + 1] = image[ix_ip, iy_ip]
```

Comment:
What is all this?

Reply:
This is just an attempt to vectorize the codes to remove the loop such that the code speeds up. If you look the `initial_direction_finding` function above this and compare it across this commit you will see I have removed the loop in it which decreased the example run time by almost 20secs. I am trying to do that for this also but this is tricky. I will try to fix it if possible otherwise I will just remove this block.